lib/src/main/kotlin/com/swmansion/starknet/data/types/conversions/ConvertibleToFeltList.kt

```diff
@@ -0,0 +1,14 @@
+package com.swmansion.starknet.data.types.conversions
+
+import com.swmansion.starknet.data.types.Felt
+
+/**
+ * Implementers of this interface support conversions to
+ * the list of Felts.
+ */
+interface ConvertibleToFeltList {
```

Comment:
I don't like the name of it. Maybe `StarknetSerializable`/`CalldataSerializable`? In general starknet might support some other data types in the future, I don't want to tie it to felt. We can name `toFeltList` to `toCalldata`.

Reply:
`CalldataSerializable` is definitely a no, since it can be used for other things. Take a `StarknetCurveSignature` for example where it felt list representation is used as a signature.

Reply:
At first I wanted to do something like `CalldataConvertible` and use it in dataclasses utilizing calldata, but it was more problematic than useful: You'd have to somehow convince the compiler that the (very ergonomic) `List<Felt>` is also `CalldataConvertible` and do it in such way, that it also works in Java.

And indicating that something is convertible to Calldata when in fact you mean it's just convertible to Felt is just misleading.

Reply:
Signature is also in calldata format.

Reply:
It is a calldata for verify function

Reply:
Technically yes, but we do have a separate type for it and I want to follow that. If we want to treat Signature the same as Calldata, we'd have to `Calldata` typing instead.

Reply:
Right, just don't support CalldataSerializable with our signature class in crypto module